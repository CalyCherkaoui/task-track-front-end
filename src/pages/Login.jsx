/* eslint-disable no-console */
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { login } from '../actions/authentication';
// import { Redirect } from 'react-router-dom';

const Login = () => {
  const dispatch = useDispatch();
  const [username, setUsername] = useState();
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(login(username, email, password));
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="login_username_input">
        Username:
        <input
          id="login_username_input"
          type="text"
          name="username"
          placeholder="Enter your Username"
          onChange={(e) => setUsername(e.target.value)}
          required
        />
      </label>
      <label htmlFor="login_email_input">
        Email:
        <input
          id="login_email_input"
          type="email"
          name="email"
          placeholder="Enter your email!"
          onChange={(e) => setEmail(e.target.value)}
          required
        />
      </label>
      <label htmlFor="login_password_input">
        Password:
        <input
          id="login_password_input"
          type="password"
          name="password"
          placeholder="Enter your password!"
          onChange={(e) => setPassword(e.target.value)}
          required
        />
      </label>
      <button
        type="submit"
      >
        Log me in!
      </button>
    </form>
  );
};

export default Login;
