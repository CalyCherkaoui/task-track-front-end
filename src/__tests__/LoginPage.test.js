import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store';
import Login from '../pages/Login';

describe(
  'Renders correctly Login page component',
  () => {
    let container;
    beforeEach(() => {
      container = shallow(
        <Provider store={store}>
          <Router>
            <Login />
          </Router>
        </Provider>,
      );
    });

    it(
      'Contains \'Log me in\' button',
      () => {
        expect(container.find('Log me in')).toBeTruthy();
      },
    );

    it(
      'Matches Login snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <Login />
            </Router>
          </Provider>,
        );

        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
