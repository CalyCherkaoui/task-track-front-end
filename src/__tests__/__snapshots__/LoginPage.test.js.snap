// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly Login page component Matches Login snapshot 1`] = `
<div
  className="form_wrapper box_flex_col_centered p-4"
>
  <div
    className="g-0 shadowed_small container"
  >
    <div
      className="form_header_wrapper g-0 row"
    >
      <div
        className="box_flex_col_centered g-0 col"
      >
        <p
          className="form_header_text"
        >
          Login
        </p>
      </div>
    </div>
    <div
      className="form_form_wrapper g-0 row"
    >
      <div
        className="box_flex_col_centered py-4 col"
      >
        <form
          className="form_form px-3"
          onSubmit={[Function]}
        >
          <label
            className="form_label box_flex_col_centered"
            htmlFor="login_username_input"
          >
            <input
              className="form_input"
              id="login_username_input"
              name="username"
              onChange={[Function]}
              placeholder="Enter your username"
              required={true}
              type="text"
            />
          </label>
          <label
            className="form_label box_flex_col_centered"
            htmlFor="login_email_input"
          >
            <input
              className="form_input"
              id="login_email_input"
              name="email"
              onChange={[Function]}
              placeholder="Enter your email!"
              required={true}
              type="email"
            />
          </label>
          <label
            className="form_label box_flex_col_centered"
            htmlFor="login_password_input"
          >
            <input
              className="form_input"
              id="login_password_input"
              name="password"
              onChange={[Function]}
              placeholder="Enter your password!"
              required={true}
              type="password"
            />
          </label>
          <button
            className="form_submit"
            type="submit"
          >
            Log me in!
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
